scipy/stats/_stats_mstats_common.py

```diff
@@ -146,6 +146,10 @@ def linregress(x, y=None, alternative='two-sided'):
     if x.size == 0 or y.size == 0:
         raise ValueError("Inputs must not be empty.")
 
+    if np.amax(x) == np.amin(x) and len(x) > 1:
```

Comment:
We should not need the length check since the min would be max if there is just a single element.
```suggestion
    if np.amax(x) == np.amin(x):
```

Reply:
I added a length check because the case where there is only a single element is already tested for in `scipy.stats.tests.test_morestats::TestProbplot::test_array_of_size_one` and I didn't want to change existing behaviours